packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -0,0 +1,562 @@
+///    plane's viewport painting coordinates** is also done such that the child
+///    in the center of that plane will be mostly untransformed with children
+///    above and below it being transformed more as the angle increases.
+class RenderListWheelViewport
+    extends RenderBox
+    with ContainerRenderObjectMixin<RenderBox, _ListWheelParentData>,
+         RenderBoxContainerDefaultsMixin<RenderBox, _ListWheelParentData>
+    implements RenderAbstractViewport {
+  RenderListWheelViewport({
+    @required ViewportOffset offset,
+    @required double diameterRatio,
+    @required double perspective,
+    @required double itemExtent,
+    @required bool clipToSize,
+    @required bool renderChildrenOutsideViewport,
+    List<RenderBox> children,
+  }) : assert(offset != null),
+       _offset = offset,
+       _diameterRatio = diameterRatio,
+       _perspective = perspective,
+       _itemExtent = itemExtent,
+       _clipToSize = clipToSize,
+       _renderChildrenOutsideViewport = renderChildrenOutsideViewport {
+    addAll(children);
+  }
+
+  /// The associated ViewportOffset object for the viewport.
+  ViewportOffset get offset => _offset;
+  ViewportOffset _offset;
+  set offset(ViewportOffset value) {
+    assert(value != null);
+    if (value == _offset)
+      return;
+    if (attached)
+      _offset.removeListener(_hasScrolled);
+    _offset = value;
+    if (attached)
+      _offset.addListener(_hasScrolled);
+    markNeedsLayout();
+  }
+
+  /// {@template flutter.rendering.wheelList.diameterRatio}
+  /// A ratio between the diameter of the cylinder and the viewport's size
+  /// in the main axis.
+  ///
+  /// A value of 1 means the cylinder has the same diameter as the viewport's
+  /// size.
+  ///
+  /// A value smaller than 1 means items at the edges of the cylinder are
+  /// entirely contained inside the viewport.
+  ///
+  /// A value larger than 1 means angles less than +/- [math.pi] / 2 from the
+  /// center of the cylinder are visible.
+  ///
+  /// The same number of children will be visible in the viewport regardless of
+  /// the [diameterRatio]. The number of children visible is based on the
+  /// viewport's length along the main axis divided by the children's
+  /// [itemExtent]. Then the children are evenly distributed along the visible
+  /// angles up to +/- [math.pi] / 2.
+  ///
+  /// Just as it's impossible to stretch a paper to cover the an entire
+  /// half of a cylinder's surface where the cylinder has the same diameter
+  /// as the paper's length, choosing a [diameterRatio] smaller than [math.pi]
+  /// will leave same gaps between the children.
+  /// {@endtemplate}
+  double get diameterRatio => _diameterRatio;
+  double _diameterRatio;
+  set diameterRatio(double value) {
+    assert(value != null);
+    assert(
+      value > 0,
+      "You can't set a diameterRatio of 0. It would imply a cylinder of 0 "
+      'diameter in which case nothing will be drawn');
```

Comment:
all these asserts should be in the constructor, otherwise the constructor is a way to smuggle values in.

Reply:
Done